src/extension.ts

```diff
@@ -26,20 +26,31 @@ export interface ScopeInfo {
   configurationTarget: boolean;
 }
 
+export interface XMLFileAssociation {
+  systemId: string,
+  pattern: string;
+}
+
+export interface XMLExtensionApi {
```

Comment:
As it's an API, add Javadoc to explain this API. I think we should write in the description how to consume this API by given a sample in the comments.

Reply:
I placed the comments on top of each API below

Reply:
Thanks but my idea is to write a sample code by getting extension and consume the API, so I think the Javadoc should mention this piece of code: 

```ts
const ext = await vscode.extensions.getExtension('redhat.vscode-xml').activate();
ext.addXMLCatalogs...
```

Reply:
I see! let me edit that.